src/libYARP_OS/src/Type.cpp

```diff
@@ -7,60 +7,185 @@
  */
 
 #include <yarp/os/Type.h>
+
 #include <yarp/os/Property.h>
-#include <yarp/os/Log.h>
+#include <yarp/os/Searchable.h>
+#include <yarp/os/Value.h>
 
 using namespace yarp::os;
 
-Type::Type() {
-    prop = nullptr;
-}
+class Type::Private
+{
+public:
+    Private() :
+            prop(nullptr)
+    {
+    }
 
-Type::Type(const Type& alt) {
-    prop = nullptr;
-    name = alt.name;
-    name_on_wire = alt.name_on_wire;
-    if (alt.prop) {
-        writeProperties();
-        *prop = *(alt.prop);
+    Private(const Private& rhs) :
+            prop(nullptr),
+            name(rhs.name),
+            name_on_wire(rhs.name_on_wire)
+    {
+        if (rhs.prop) {
+            writeProperties();
+            *prop = *(rhs.prop);
+        }
+    }
+
+    Private(Private&& rhs) noexcept :
+            prop(rhs.prop),
+            name(std::move(rhs.name)),
+            name_on_wire(std::move(rhs.name_on_wire))
+    {
+        rhs.prop = nullptr;
     }
-}
 
-Type::~Type() {
-    if (prop) {
+    ~Private()
+    {
         delete prop;
```

Comment:
I would add
```
if (prop)  {
    delete prop;
    prop = nullptr;
}
```
Or even better shall we move to `unique_ptr` ? :heart_eyes:

Reply:
:-1: for the extra if and assignment to `nullptr` in the destructor... they are completely useless, `delete` will not delete a `nullptr`, and after the destruction `prop` is no longer accessible, therefore no need to assign it.

:+1: for unique_ptr, but that's for another pr...